test/value_ref.cpp

```diff
@@ -323,6 +323,12 @@ class value_ref_test
             array b{std::move(a), string()};
             BOOST_TEST(a.empty());
         }
+        {
+            string a = "abcdefghijklmnopqrstuvwxyz";
+            BOOST_TEST(!a.empty());
+            array b{a, string()};
```

Comment:
I hate this form of initialization. I prefer
```
array b = { a, string() };
```
or
```
array b( { a, string() } );
```

Reply:
Meh, it's the style we use in all the other tests. If you want, I can change them all

Reply:
No I'm just grumbling. The user reported a const string though, should we test that too?

Reply:
I added a test for `string const`